components/vision.py

```diff
@@ -39,15 +53,28 @@ def __init__(
         name: str,
         # Position of the camera relative to the center of the robot
         pos: Translation3d,
-        # The camera rotation.
+        # The camera rotation at its neutral position (ie centred).
         rot: Rotation3d,
+        servo_id: int,
+        encoder_id: int,
         field: wpilib.Field2d,
         data_log: wpiutil.log.DataLog,
         chassis: ChassisComponent,
     ) -> None:
         self.camera = PhotonCamera(name)
-        self.robot_to_camera = Transform3d(pos, rot)
-        self.camera_to_robot = self.robot_to_camera.inverse()
+        # Assuming channel is 0 for encoder
+        self.encoder = wpilib.DutyCycleEncoder(encoder_id, math.tau, 0)
+        # Offset of encoder in radians when facing forwards (the desired zero)
+        # To find this value, manually point the camera forwards and record the encoder value
+        # This has nothing to do with the servo - do it by hand!!
+        self.encoder_offset = Rotation2d(0.2)
+
+        # To find the servo offset, command the servo to neutral in test mode and record the encoder value
+        self.servo_offset = Rotation2d(0.1)
```

Comment:
Feels odd that we're dependency injecting the other extrinsics but not these. Let's be consistent?

Reply:
@auscompgeek  @james-ward I am a fan of removing the dependency injection from this class now that we will likely only run a single camera. I think it should come from a separate pr so we can get this in asap. I wanted to test it one last time after #56 before sending this in

Reply:
Dependency inject all the things!